### PR TITLE
Add card functions

### PR DESCRIPTION
**Pull request by benadida**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -32,6 +32,13 @@ const App: React.FC = () => {
     setPrecinct('')
     setBallot('')
   }
+  const programCard = () => {
+    fetch('/card/write', {
+      method: 'post',
+      body: JSON.stringify({ code: 'VX.' + ballot + '.' + precinct }),
```
Comment (beausmith) [inline src/App.tsx]: `ballot` and `precinct` are in transposed compared to activation codes used in the bmd code: `VX.precinct-23.12D`

Also use template tags > concatenated strings 😄  

```
JSON.stringify({ code: `VX.${ballot}.${precinct}` })
```

File: src/App.tsx [hunk 2]
```diff
@@ -32,6 +32,13 @@ const App: React.FC = () => {
     setPrecinct('')
     setBallot('')
   }
+  const programCard = () => {
+    fetch('/card/write', {
```
Comment (beausmith) [inline src/App.tsx:36]: Would be great to call `reset()` on success. 

```js
fetch(…)
.then(res => res.json())
.then(response => reset(response)) // Update reset() to have a "successful activation" reset vs manual reset
.catch(error => console.error('Error:', error));
```

**Review by beausmith — COMMENTED** [review]
Some comments. Ping me again when build is 💚

File: src/App.tsx (hunk 1 above)
Comment (benadida) [reply src/App.tsx]: 👍

File: .gitignore [hunk 3]
```diff
@@ -23,4 +23,5 @@ yarn-debug.log*
 yarn-error.log*
 
 # python
-*.pyc
\ No newline at end of file
+*.pyc
+.coverage
```
Comment (beausmith) [inline .gitignore:27]: nit: newline

**Review by beausmith — APPROVED** [review]
Approving because I'd consider this beta and I'm trusting that you know what you're doing. However until I can run this all locally, I'm gonna consider it to be in beta. I know you mentioned that there are some differences between running this on a mac… but I'm sure there must be a way. 

If I can't run the python server locally, I can't click the "Program Card" button without an error because the app can't reach `/card/write`.

Additionally, if both apps are running on port `3000` there is a conflict.

Here's success and errors trying to get python server running locally…

```
beau at cleaver in ~/Development/VotingWorks/bas/server on add-card-functions
$ pipenv install
Creating a virtualenv for this project…
Pipfile: /Users/beau/Development/VotingWorks/bas/server/Pipfile
Using /usr/local/opt/python/libexec/bin/python (3.7.3) to create virtualenv…
⠦ Creating virtual environment...Using base prefix '/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7'
New python executable in /Users/beau/.local/share/virtualenvs/server-vvnjb66D/bin/python3.7
Also creating executable in /Users/beau/.local/share/virtualenvs/server-vvnjb66D/bin/python
Installing setuptools, pip, wheel...
done.
Running virtualenv with interpreter /usr/local/opt/python/libexec/bin/python

✔ Successfully created virtual environment! 
Virtualenv location: /Users/beau/.local/share/virtualenvs/server-vvnjb66D
Installing dependencies from Pipfile.lock (9b05e9)…
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 16/16 — 00:00:02
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.

beau at cleaver in ~/Development/VotingWorks/bas/server on add-card-functions
$ pipenv shell
Launching subshell in virtual environment…
bash: prompt_git: command not found

beau at cleaver in ~/Development/VotingWorks/bas/server
$  . /Users/beau/.local/share/virtualenvs/server-vvnjb66D/bin/activate
bash: prompt_git: command not found
(server) 
beau at cleaver in ~/Development/VotingWorks/bas/server
$ make run
FLASK_APP=basserver.core python -m flask run --port 3000
 * Serving Flask app "basserver.core"
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:3000/ (Press CTRL+C to quit)
127.0.0.1 - - [02/May/2019 22:49:57] "GET / HTTP/1.1" 404 -
127.0.0.1 - - [02/May/2019 22:49:57] "GET /favicon.ico HTTP/1.1" 404 -
^C
bash: prompt_git: command not found
(server) 
beau at cleaver in ~/Development/VotingWorks/bas/server
$ 
```

**Comment by benadida** [comment]
@beausmith I don't think we should consider running the python server on Mac a blocking feature for this being non-beta, given that I don't even know how to get the smart-card stack working on the Mac just yet (and not for lack of trying either), and we don't expect this to ever run on a Mac. Of course, if we find a way to also make it work, I have no objection, but that doesn't seem like a priority given how long it is likely to take to solve (and to maintain.) The smart card stacks are just very finicky.

Important clarification: the React dev server and Python server are not meant to be running at the same time. For the Python server, `yarn build` and then run the Python server

I do think in React dev you should be able to click "program card" and get a reasonable behavior so that all behavior (including tests) can ensue without having the Python server running. Basically, I want to set things up so that everything works without the Python server except the smart card operations, and even then it's still possible to see every flow in the UI. I've created #8 to address this.

**Pull request merged**